frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailsColumns.tsx

```diff
@@ -0,0 +1,112 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { Skeleton } from '@patternfly/react-core'
+import { ListItems } from '../../../../ui-components'
+import { generatePath, Link } from 'react-router-dom-v5-compat'
+import { NavigationPath } from '../../../../NavigationPath'
+import { ExternalLinkAltIcon } from '@patternfly/react-icons'
+import { TFunction } from 'react-i18next'
+
+// Any resources that are related to Vapb then add these rows
+export const addRowsForHasVapb = (
+  cols: ListItems[],
+  hasVapb: boolean,
+  loading: boolean,
+  vapbItems: any[] | null | undefined,
+  apiGroup: string,
+  clusterName: string,
+  name: string
+): ListItems[] => {
+  if (!hasVapb) {
+    return cols
+  }
+  // Loading to fetch VAPB
+  if (!vapbItems) {
+    cols.push({
+      key: 'Validating Admission Policy Binding',
+      value: <Skeleton width="100%" screenreaderText="Fetching ValidatingAdmissionPolicyBinding" />,
+    })
+  } else if (vapbItems && vapbItems.length > 0 && !loading) {
+    const vapbName = apiGroup === 'constraints.gatekeeper.sh' ? `gatekeeper-${name}` : name + '-binding'
+    cols.push({
+      key: 'Validating Admission Policy Binding',
+      value: (
+        <Link
+          to={generatePath(NavigationPath.discoveredPolicyDetails, {
+            clusterName,
+            apiVersion: 'v1',
+            apiGroup: 'admissionregistration.k8s.io',
+            kind: 'ValidatingAdmissionPolicyBinding',
+            templateName: vapbName,
+            templateNamespace: null,
+          })}
+          target="_blank"
+        >
+          {vapbName} <ExternalLinkAltIcon style={{ verticalAlign: '-0.125em', marginLeft: '8px' }} />
+        </Link>
+      ),
+    })
+  } else if (vapbItems && vapbItems.length == 0 && !loading) {
+    cols.push({
+      key: 'Validating Admission Policy Binding',
+      value: '-',
+    })
+  }
+
+  return cols
```

Comment:
Do you need to return the columns in these functions since it looks like you're modifying the cols array in place with `.push`?